osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs

```diff
@@ -133,6 +133,64 @@ public List<TimedDifficultyAttributes> CalculateTimed([NotNull] IEnumerable<Mod>
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>Per-skill array where each value represents difficulty of this section in certain skill.</returns>
+        public IEnumerable<double[]> CalculateSectionDifficulties(CancellationToken cancellationToken = default)
+            => CalculateSectionDifficulties(Array.Empty<Mod>(), cancellationToken);
+
+        /// <summary>
+        /// Calculates the difficulty of section of the beatmap using a specific mod combination.
+        /// </summary>
+        /// <param name="mods">The mods that should be applied to the beatmap.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>Per-skill array where each value represents difficulty of this section in certain skill.</returns>
+        public IEnumerable<double[]> CalculateSectionDifficulties([NotNull] IEnumerable<Mod> mods, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            preProcess(mods, cancellationToken);
+
+            var skills = CreateSkills(Beatmap, playableMods, clockRate);
+            StrainSkill[] relevantSkills = GetRelevantSkills(skills);
+            var hitObjects = getDifficultyHitObjects();
+
+            if (!hitObjects.Any())
+                return Enumerable.Empty<double[]>();
+
+            // Add sections before first object to preserve correct bounds
+            foreach (var skill in relevantSkills)
+                skill.AddEmptySections(hitObjects.First().StartTime, Beatmap.HitObjects.First().StartTime / clockRate);
+
+            foreach (var hitObject in hitObjects)
+            {
+                foreach (var skill in relevantSkills)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    skill.Process(hitObject);
+                }
+            }
+
+            // Add sections up to the end time of last object to preserve correct bounds
+            foreach (var skill in relevantSkills)
+                skill.AddEmptySections(Beatmap.HitObjects.Last().GetEndTime() / clockRate);
+
+            var strainsForSkills = relevantSkills
+                .Select(skill => skill.GetCurrentStrainPeaks().ToArray());
+
+            return strainsForSkills;
+        }
+
+        /// <summary>
+        /// Filters relevant skills that would be used to represent difficulty of the section.
+        /// </summary>
+        /// <remarks>
+        /// IMPORTANT NOTE: for now it's using <see cref="StrainSkill"/> to receive section values, consider changing it in the future.
+        /// Also, this function should also be changed if filtering skills is no longer enough to get desired values.
+        /// </remarks>
+        public virtual StrainSkill[] GetRelevantSkills(Skill[] skills) => skills.OfType<StrainSkill>().ToArray();
```

Comment:
this can't exist

Reply:
How else can I filter the skills to get only those that I need
Specifically this means that I need to filter-out SliderlessAim from STD Difficulty Calculation

Reply:
why are you filtering skills at all would be my primary question here?

Reply:
> why are you filtering skills at all would be my primary question here?

Aim and SliderlessAim are the same skill
leaving them both means that aim difficulty is added 2 times

Reply:
I can add `public virtual bool IsRelevant => true; // Is skill relevant to star rating` field to the `Skill` class
But it's +- the same as current solution, just written in different way